src/components/AppRepoList/AppRepoButton.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from "react";
+import * as Modal from "react-modal";
+import { Redirect } from "react-router";
+
+interface IAppRepoFormProps {
+  name: string;
+  url: string;
+  message?: string;
+  redirectTo?: string;
+  install: (name: string, url: string) => Promise<any>;
+  update: (values: { name?: string; url?: string }) => void;
+  onAfterInstall?: () => Promise<any>;
+}
+
+export const AppRepoForm = (props: IAppRepoFormProps) => {
+  const { name, url, update, install, onAfterInstall } = props;
+  const handleInstallClick = async () => {
```

Comment:
When a component has several methods to manage interactions from users, I'd rather use a normal class syntax instead of creating functions inside the "render method"

Also, for me `install` seems to be a method that should be executed in a container instead of on a presentational component.